tests/transformers/test_numerical.py

```diff
@@ -33,6 +33,194 @@ def test_fit(self):
+        big_numbers = [10**exponents[i] for i in range(10)]
+        data = np.array(big_numbers)
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan='nan', rounding='auto')
+        transformer.fit(data)
+
+        # Asserts
+        expected = -min(exponents) if min(exponents) > 0 else None
+        assert transformer.rounding_digits == expected
+
+    def test_fit_rounding_auto_max_decimals(self):
+        """Test fit rounding parameter with ``'auto'``
+
+        If the ``rounding`` parameter is set to ``'auto'``,
+        fit should learn the ``rounding_digits`` to be the max
+        number of decimal places seen in the data. The biggest
+        amount of decimals that can be accurately compared with
+        floats is 15.
+
+        Input:
+        - Array with a value that has over 15 decimals.
+        """
+        # Setup
+        data = np.array([0.000000000000001])
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan='nan', rounding='auto')
+        transformer.fit(data)
+
+        # Asserts
+        assert transformer.rounding_digits is None
+
+    def test_reverse_transform_rounding_none(self):
+        """Test ``reverse_transform`` when ``rounding`` is ``None``
+
+        The data should not be rounded at all.
+        """
+        # Setup
+        data = np.random.random(10)
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan=None)
+        transformer.rounding_digits = None
+        result = transformer.reverse_transform(data)
+
+        # Assert
+        assert np.array_equal(result, data)
+
+    def test_reverse_transform_rounding_positive_int(self):
+        """Test ``reverse_transform`` when ``rounding`` is a positive int
+
+        The data should round to the maximum number of decimal places
+        set in the ``rounding_digits`` value.
+
+        Input:
+        - Array with decimals
+
+        Output:
+        - Same array rounded to the provided number of decimal places
+        """
+        # Setup
+        data = np.array([1.1111, 2.2222, 3.3333, 4.44444, 5.555555])
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan=None)
+        transformer.rounding_digits = 2
+        result = transformer.reverse_transform(data)
+
+        # Assert
+        expected_data = np.array([1.11, 2.22, 3.33, 4.44, 5.56])
+        assert np.array_equal(result, expected_data)
+
+    def test_reverse_transform_rounding_negative_int(self):
+        """Test ``reverse_transform`` when ``rounding`` is a negative int
+
+        The data should round to the number set in the ``rounding_digits``
+        attribute.
+
+        Input:
+        - Array with with larger numbers
+
+        Output:
+        - Same array rounded to the provided number of 0s
+        """
+        # Setup
+        data = np.array([2000, 120, 3100, 40100])
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan=None)
```

Comment:
Since this is an integer test case, the `dtype` should be `int`, to guarantee that the output is an integer array.
A separated test case should be created to also round values to negative number of decimals but keep it a `float` array